pkg/apis/core/v1alpha1/uicomponentlocation_types.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+Copyright 2021 The Tilt Dev Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// Specifies where to put a UI Component
+type UIComponentLocation struct {
+	// which resource to place the component on
+	Resource *UIComponentLocationResource `json:"resource"`
+	// If there are multiple components in the same location, they will be
+	// located in order from lowest to highest. Ties are broken arbitrarily.
+	Order int `json:"order"`
```

Comment:
I'm not a big fan of this solution but it seems good enough for now.

It's especially a problem if we have multiple extensions defining components, all trampling on each others' orders.

If someone has a better idea off the top of their head, I'm eager to hear it!

Reply:
[Server-enforced fractional indexing](https://www.figma.com/blog/realtime-editing-of-ordered-sequences/#fractional-indexing)? (I'm joking)

This seems fine to me -- I could eventually foresee splitting the widget + layout specs, so e.g. an extension could register a bunch of element/widgets and the user could see them all via some sort of gallery and drag around to form their layout, but that's waaaay overkill at this point 😛